nilearn/maskers/surface_labels_masker.py

```diff
@@ -0,0 +1,463 @@
+
+    Attributes
+    ----------
+    n_elements_ : :obj:`int`
+        The number of discrete values in the mask.
+        This is equivalent to the number of unique values in the mask image,
+        ignoring the background value.
+    """
+
+    def __init__(
+        self,
+        labels_img,
+        labels=None,
+        background_label=0,
+        smoothing_fwhm=None,
+        standardize=False,
+        standardize_confounds=True,
+        detrend=False,
+        high_variance_confounds=False,
+        low_pass=None,
+        high_pass=None,
+        t_r=None,
+        memory=None,
+        memory_level=1,
+        verbose=0,
+        reports=True,
+        cmap="inferno",
+        clean_args=None,
+    ):
+        self.labels_img = labels_img
+        self.labels = labels
+        self.background_label = background_label
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.standardize_confounds = standardize_confounds
+        self.high_variance_confounds = high_variance_confounds
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+        self.memory_level = memory_level
+        self.verbose = verbose
+        self.reports = reports
+        self.cmap = cmap
+        self.clean_args = clean_args
+        self._shelving = False
+        # content to inject in the HTML template
+        self._report_content = {
+            "description": (
+                "This report shows the input surface image overlaid "
+                "with the outlines of the mask. "
+                "We recommend to inspect the report for the overlap "
+                "between the mask and its input image. "
+            ),
+            "n_vertices": {},
+            "number_of_regions": 0,
+            "summary": {},
+        }
+
+    @property
+    def _labels_data(self):
+        """Return data of label image concatenated over hemispheres."""
+        return np.concatenate(list(self.labels_img.data.parts.values()))
+
+    def fit(self, img=None, y=None):
+        """Prepare signal extraction from regions.
+
+        Parameters
+        ----------
+        img : :obj:`~nilearn.surface.SurfaceImage` object or None, default=None
+            This parameter is currently unused.
+
+        y : None
+            This parameter is unused.
+            It is solely included for scikit-learn compatibility.
+
+        Returns
+        -------
+        SurfaceLabelsMasker object
+        """
+        del img, y
+
+        all_labels = set(self._labels_data.ravel())
+        all_labels.discard(self.background_label)
+        self._labels_ = list(all_labels)
+
+        self.n_elements_ = len(self._labels_)
+
+        if self.labels is None:
+            self._label_names_ = [str(label) for label in self._labels_]
+        else:
+            self._label_names_ = [self.labels[x] for x in self._labels_]
```

Comment:
@mtorabi59 
I am keeping most of the fitted attributes private so we can later improve how to better align how the different label maskers (nifti and surface) work with missing labels and all the other issues we uncovered with label maskers.

Reply:
@Remi-Gau sounds good!